interfaces/vault_autounseal/v0/README.md

```diff
@@ -0,0 +1,94 @@
+# `vault-autounseal`
+
+## Usage
+
+This relation interface describes the expected behavior of a charm that integrates with the Vault Transit backend to autounseal a vault over the `vault-autounseal` relation.
+
+## Data Flow
+
+```mermaid
+flowchart LR
+  Provider --
+    address
+    key_name
+    credentials_secret_id
+    ca_certificate
+    --> Requirer
+  Requirer
+```
+
+## Behavior
+
+Both the Requirer and the Provider need to adhere to criteria to be considered compatible with the interface. Specifically, the Provider has some operations to perform, and data to provide to the Requirer. The Requirer on the other hand, must configure Vault to autounseal based on the data provided by the Provider.
+
+### Provider
+
+The Provider is expected to
+
+- create a `transit` backend with the path `charm-autounseal` if it does not already exist
+- create an encryption key for the requirer called `charm-autounseal/keys/${relation_id}` in the `charm-autounseal` backend
+- create a policy for the requirer that provides `update` capabilities on the encryption key
+- create an AppRole for this policy, and a secret for the AppRole, and store both the role ID and the secret ID in a Juju secret
+  - The secret should contain at least two key-value pairs: `role-id`, and `secret-id`
+  - The AppRole should allow the creation of periodic tokens, so that Vault can indefinitely renew the token.
+- provide the following to the Requirer in the databag
+  - the URL of the Vault (used in the Vault config for the `address` value)
+  - the key name of the transit key used for autounseal
+  - the Juju secret ID which contains the Role ID and Secret ID of the AppRole
+  - the CA certificate that the Requirer should use to validate its certificate.
+
+### Requirer
+
+The Requirer is expected to
+
+- retrieve the Role ID and Secret ID from the Juju secret that was identified by the Requirer
+- create a new token
+- configure Vault using a `seal "transit"` stanza and in the Vault config
+  - set the `address` to the address provided in the databag
+  - set the `key_name` to the key name provided in the databag
+  - set the `mount_path` to `charm-autounseal`
+  - set the token to the one created from the AppRole credentials
+  - store the CA certificate locally, and set the CA certificate path
+
+## Relation Data
+
+[\[Pydantic Schema\]](./schema.py)
+
+### Example
+
+```yaml
+provider:
+  app:
+    address: https://10.152.183.217:8200
+    key_name: 2
```

Comment:
Is this a good key_name example?

Reply:
It is. The full path currently looks something like  `mymount/keys/1`.

For now, we're just keying off of the relation ID, but we could add some prefix or suffix.